File: MCRRTImp/src/main/java/lab/mars/MCRRTImp/Vector2BasedImp/CircleObstacle.java
package lab.mars.MCRRTImp.Vector2BasedImp;


import lab.mars.MCRRTImp.Vector2BasedImp.Vector2;
import lab.mars.RRTBase.Obstacle;

public class CircleObstacle implements Obstacle<Vector2> {

    public double radius;


    public Vector2 origin;

    public CircleObstacle(double x, double y, double radius) {
        this.radius = radius;
        this.origin = new Vector2(x, y);
    }

    @Override
    public boolean contains(Vector2 o) {
        return o.distance2(origin) < radius * radius;
    }
}

File: MCRRTImp/src/test/kotlin/infrastructure/Vector2BasedImp/CircleObstacle.kt
package infrastructure.Vector2BasedImp


import lab.mars.MCRRTImp.base.Obstacle

class CircleObstacle(x: Double, y: Double, var radius: Double) : Obstacle<Vector2> {


    var origin: Vector2 = Vector2(x, y)

    override operator fun contains(o: Vector2): Boolean {
        return o.distance2(origin) < radius * radius
    }
}
